internal/provider/data_source_aws_transit_gateway_attachment.go

```diff
@@ -0,0 +1,119 @@
+package provider
+
+import (
+	"context"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceAwsTransitGatewayAttachment() *schema.Resource {
+	return &schema.Resource{
+		Description: "The AWS Transit gateway attachment data source provides information about an existing Transit gateway attachment.",
+		ReadContext: dataSourceAwsTransitGatewayAttachmentRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &tgwDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn_id": {
+				Description:      "The ID of the HashiCorp Virtual Network (HVN).",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"transit_gateway_attachment_id": {
+				Description:      "The ID of the Transit gateway attachment.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// Optional inputs
+			"state": {
+				Description: "The state of the Transit gateway attachment.",
```

Comment:
I think if we keep this approach, we need to list out the possible states that the user can pick and validate them on the input.

Reply:
Yea, I was waiting to see where we landed with the interface, but will make sure to address documentation, validation, and diffing once we make a choice.